util/string.c

```diff
@@ -3,7 +3,7 @@
 #include "string.h"
 #include <stdint.h>
 
-#if defined(__GNUC__) && !defined(__clang__)
+#if defined(__GNUC__) && !defined(__llvm__)
```

Comment:
TIL this is defined by Clang... kinda the wrong one to use here but it works and removes the diff to upstream

Reply:
Yeah I also hadn't seen that before but used the upstream version to reduce the diff.